File: Day-01/code/py/answer.py
import sys

with open(sys.argv[1], 'r') as file:
    data = file.read().strip()

left = []
right = []

for line in data.split('\n'):
    nl, nr = line.split('   ')
    left.append(int(nl))
    right.append(int(nr))

sorted_left = sorted(left)
sorted_right = sorted(right)

p1 = [abs(nl - sorted_right[i]) for i, nl in enumerate(sorted_left)]
print(f"Part 1: {sum(p1)}")

c = {}
for n in right:
    c[n] = c.get(n, 0) + 1

p2 = [n * c.get(n, 0) for n in left]
print(f"Part 2: {sum(p2)}")

File: Day-01/code/js/answer.js
const fs = require('fs');

const data = fs.readFileSync(process.argv[2], 'utf8').trim();

let left = [];
let right = [];

data.split('\n').forEach(line => {
    let [nl, nr] = line.split('   ');
    left.push(parseInt(nl));
    right.push(parseInt(nr));
});

let sortedLeft = left.slice().sort((a, b) => a - b);
let sortedRight = right.slice().sort((a, b) => a - b);

let p1 = sortedLeft.map((nl, i) => Math.abs(nl - sortedRight[i]));
console.log(`Part 1: ${p1.reduce((a, b) => a + b, 0)}`);

let c = {};

for (let n of right) {
    c[n] = (c[n] || 0) + 1;
}

let p2 = left.map(n => n * (c[n] || 0));
console.log(`Part 2: ${p2.reduce((a, b) => a + b, 0)}`);
